tensorrt_llm/serve/visual_gen_utils.py

```
@@ -90,21 +90,107 @@ def _merge_extra_params(
+    # audio: no signature sniffing (sniff_media_kind detects only image/video);
+    # the consuming pipeline validates the audio codec in its worker.
+    if media_storage_path is None:
+        raise ValueError(f"media_storage_path is required to store the {modality}_reference.")
+    ref_path = os.path.join(media_storage_path, ref_id)
+    with open(ref_path, "wb") as f:
+        f.write(payload)
+    return ref_path
+
+
+def _build_reference_list(
+    value, *, modality: str, data_field: str, ref_cls, id: str, media_storage_path: Optional[str]
```

Ruff (0.16.1)

[error] Function argument `id` is shadowing a Python builtin

(A002)